RecoEgamma/EgammaElectronAlgos/src/GsfElectronAlgo.cc

```diff
@@ -609,6 +609,12 @@ void GsfElectronAlgo::calculateShowerShape( const reco::SuperClusterRef & theClu
     showerShape.hcalTowersBehindClusters = generalData_->hcalHelperPflow->hcalTowersBehindClusters(*theClus) ;
     showerShape.hcalDepth1OverEcalBc = generalData_->hcalHelperPflow->hcalESumDepth1BehindClusters(showerShape.hcalTowersBehindClusters)/theClus->energy() ;
     showerShape.hcalDepth2OverEcalBc = generalData_->hcalHelperPflow->hcalESumDepth2BehindClusters(showerShape.hcalTowersBehindClusters)/theClus->energy() ;
+    // FIXME this should get its own flag, but nevermind for now
+    if (showerShape.hcalDepth1OverEcal == 0 && showerShape.hcalDepth2OverEcal == 0) {
```

Comment:
should we do this via detecting showerShape.hcalDepth1OverEcalBc and showerShape.hcalDepth2OverEcalBc are zero? 

There are two H/E variables, cone of 0.15 and single tower behind the seed cluster. The 0.15 version could be non zero by a tiny amount even if most of the region is off.  I think it makes more sense to say "if the tower behind the electron is bad, enable mitigations".

Reply:
So another comment on what will contentious, the flagging of bad HCAL for.  I understand why you are doing it for testing, compiling the entire of CMSSW is not fun. But I really think for the final release when we are happy, we should swap it out for a flag. What we have can not be a permanent solution. Also it may be useful for that flag to be UNDEFINED, ACTIVE, INACTIVE so we know if we havent set it (aka when reading current 10_2 data) but I dont feel strongly about that.

Reply:
Ok I will change the check to hcalDepth1OverEcalBc (at a superficial look in the code it seemed to me even the method that supposedly looks behind all clusters is normally configured to look only behind the seed, and so that it doesn't matter).
About whether to add a flag or just methods, we can discuss tomorrow.

Reply:
I finally managed to finish compiling a version with the flag added, will push on github before lunch